compiler/aarch64/codegen/ControlFlowEvaluator.cpp

```diff
@@ -466,11 +471,173 @@ OMR::ARM64::TreeEvaluator::acmpeqEvaluator(TR::Node *node, TR::CodeGenerator *cg
+      startLabel->setStartInternalControlFlow();
+      pivotLabel->setEndInternalControlFlow();
+      generateLabelInstruction(cg, TR::InstOpCode::label, lookupNode, startLabel);
+
+      // We are guaranteed that the case children are sorted.
+      // Image the Z32 number lines
+      //                [------------------------|--------------------------)                                                     signed
+      //   TR::getMinSigned<TR::Int32>()         0            TR::getMaxSigned<TR::Int32>()      TR::getMaxUnsigned<TR::Int32>()
+      //                                         [--------------------------|------------------------------------)                unsigned
+      // If all the cases of the lookup are on the same 'half' of the number lines, it doesn't matter
+      // if we use JA or JG, so lets use JG.
+      // For the signed case, we better use TR::CC_GT if the numbers straddle the 0 boundary; we already do.
+      // For the unsigned case, we better use TR::CC_CS if the numbers straddle the TR::getMaxSigned<TR::Int32>() boundary. If that's
+      // the case, the highVal will be less than (in the signed sense) than lowVal.
+      //
+      bool isUnsigned;
+      int32_t lowVal  = lookupNode->getChild(lowChild)->getCaseConstant();
+      int32_t highVal = lookupNode->getChild(highChild)->getCaseConstant();
+      if (highVal < lowVal)
+         {
+         isUnsigned = true;
+         }
+      else
+         {
+         isUnsigned = false;
+         }
+
+      generateConditionalBranchInstruction(cg, TR::InstOpCode::b_cond, lookupNode, pivotLabel, isUnsigned ? TR::CC_CS : TR::CC_GT);
+
+      if (lowChild == pivot)
+         {
+         generateConditionalBranchInstruction(cg, TR::InstOpCode::b_cond, lookupNode, lookupNode->getChild(lowChild)->getBranchDestination()->getNode()->getLabel(), TR::CC_EQ);
+         generateLabelInstruction(cg, TR::InstOpCode::label, lookupNode, lookupNode->getChild(1)->getBranchDestination()->getNode()->getLabel());
+         evaluateDefaultGlRegDeps = false;
+         }
+      else
+         {
+         binarySearchCaseSpace(cg, selectorReg, lookupNode, lowChild, pivot, evaluateDefaultGlRegDeps);
+         }
+      generateLabelInstruction(cg, TR::InstOpCode::label, lookupNode, pivotLabel);
+      }
+   else
+      {
+      TR_ASSERT(pivot == lowChild - 1 && lowChild == highChild, "unexpected pivot value in binarySearchCaseSpace");
+      }
+
+   if (highChild == pivot + 1)
+      {
+      int32_t  highValue = lookupNode->getChild(highChild)->getCaseConstant();
+      generateCompareImmInstruction(cg, lookupNode, selectorReg, highValue);
+      generateConditionalBranchInstruction(cg, TR::InstOpCode::b_cond, lookupNode, lookupNode->getChild(highChild)->getBranchDestination()->getNode()->getLabel(), TR::CC_EQ);
+      generateLabelInstruction(cg, TR::InstOpCode::label, lookupNode, lookupNode->getChild(1)->getBranchDestination()->getNode()->getLabel());
```

Comment:
I wonder this should be a branch to the default path, `TR::InstOpCode::b` instead of `TR::InstOpCode::label`.
There is no branch to the default path.

Reply:
Ok, thanks! Yes, I suspect there should be a branch to the default path here.